matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/TokenChunkEventPersistor.kt

```diff
@@ -80,7 +81,26 @@ internal class TokenChunkEventPersistor @Inject constructor(
 
                     val existingChunk = ChunkEntity.find(realm, roomId, prevToken = prevToken, nextToken = nextToken)
                     if (existingChunk != null) {
-                        Timber.v("This chunk is already in the db, returns")
+                        Timber.v("This chunk is already in the db, checking if this might be caused by broken links")
```

Comment:
Can you just create a method to keep this a bit smaller

Reply:
By the way, is this still necessary with your second commit?

Reply:
I can probably do that this weekend or next week.
It's probably only needed if we want to retrospectively fix existing cases of the issue without requiring users to do a fresh initial sync to fix it. Commit two should avoid this scenario, if I haven't missed any case.

Reply:
Yeah, this should, even if there is probably another issue underneath as I don't think we should have multiple same chunks in the db :/

Reply:
We still have some issues with timeline chunk loops, like in this issue: https://github.com/vector-im/element-android/issues/5092. With some added debugging, I've seen loops starting with 4 chunks, and up to 81 chunks forming a loop (timeline chunks linking forwards and backwards indirectly to the same chunk). Seems more likely to happen while viewing a room in which spam is going on. I wonder if those loops are caused by a related bug to this one.

Reply:
it's kind of hard to replicate, but there is something wrong definitively. 
Nevermind, let's merge this to mitigate. Thanks!